docs/bc-government-api-guidelines.md

```diff
@@ -0,0 +1,209 @@
+
+- **_Use Unicode for Encoding_** – Unicode Transformation Format-8 ([UTF-8](http://unicode.org/faq/utf_bom.html#utf8-1)) is the standard encoding type for all text and textual representations of data through APIs. It must be used for all APIs published across the BC Government for both internal and external consumption
+- **_Standardize Datetime Format_** – use the ISO 8601 Standard for datetime representation in all BC Government APIs.  The standard date format is YYYY-MM-DD while timestamp format YYYY—MM-DD HH24:MI:SS.  If other formats are required due to source system limitations, convert it to the standard format within the API
+- **_Support Official Languages_** – ensure the API can return responses in both English and French.   External facing APIs must reply with content in the requested language if the backend data support it. APIs must interpret the ACCEPT-LANGUAGE HTTP header and return the appropriate content. If the header is not set, then content in both languages should be returned
+
+
+## API Documentation
+
+BC Government APIs must be published to the [BC Government API Registry](https://catalogue.data.gov.bc.ca/group/bc-government-api-registry) to be discoverable.  The documentation MVP for BC Government APIs includes everything the API does, including resources, endpoints and methods, parameters, error codes, and example requests and responses:
+
+- **_Publish OpenAPI Specification_** – OpenAPI is a machine-readable interface specification for RESTful APIs. There are open source tools (e.g., [Swagger](https://swagger.io/solutions/api-documentation/)) which can then generate human-readable documentation from this specification which avoids the need to create and maintain separate documentation
+  - Sample BC Government API Specifications can be found [here](https://catalogue.data.gov.bc.ca/dataset/bc-data-catalogue-api/resource/0b9e7d31-91ff-4146-a473-106a3b301964)
+- **_Publish code samples and test data_** – the most effective way to document the scope and functionality of an API is to publish the code and data examples used to validate it alongside the API contract
+  - Be sure to include standard and custom error response codes and interpret their specific meaning
+- **_Maintain concise documentation_** – if there is a need to extend the API documentation beyond the OpenAPI specification, this is an indication that the API is too large and/or too complex
+  - Be sure to document differences between versions of the API
+  - Include documentation on any consumption constraints such as availability, authorization and rate limiting
+- **_Gather Feedback_** – provide a clear mechanism to allow consumers to provide feedback, issue identification and enhancement requests
+
+  - The BC Government API Registry can be found here:
+[https://catalogue.data.gov.bc.ca/group/bc-government-api-registry](https://catalogue.data.gov.bc.ca/group/bc-government-api-registry)
+  - The BC Government API Github Repo can be found here:
```

Comment:
This repo remains relevant. API documentation can be added here OR teams can simply include it alongside their code (or a separate docs repo), assuming they have a public repo.

Reply:
@rustyjux - Sorry, I'm not clear if anything needs to be changed here.